classes/Product.php

```diff
@@ -1123,26 +1123,6 @@ public function validateField($field, $value, $id_lang = null, $skip = [], $huma
         return parent::validateField($field, $value, $id_lang, $skip, $human_errors);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function toggleStatus()
```

Comment:
this method is used by UpdateProductStatusHandler, shouldn't you adapt its implementation?